### PR TITLE
feat: ENS name & link format

### PR DESCRIPTION
**Pull request by borcherd**
Changelog: 
- Made function to resolve to ENS name. Takes in address and optional provider. Returns the ENS name OR null if there is no ens name found
- Updated getLinkDetails to return the senderAddress of the deposit & updated test 
- Updated structure of the link: password is now the only param hidden behind the # for the server.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -1823,6 +1824,7 @@ async function getLinkDetails({ link, provider }: interfaces.IGetLinkDetailsPara
 		depositIndex: depositIdx,
 		contractVersion: contractVersion,
 		password: password,
+		senderAddress: senderAddress,
```
Comment (Hugo0) [inline src/index.ts:1827]: idk how we missed this

File: src/index.ts [hunk 2]
```diff
@@ -1837,6 +1839,20 @@ async function getLinkDetails({ link, provider }: interfaces.IGetLinkDetailsPara
 	}
 }
 
+async function resolveToENSName({
```
Comment (Hugo0) [inline src/index.ts:1842]: I guess seeing how simple this function is, does it even make sense to have it? It's just a wrapper around ethers function.

Comment (Hugo0) [reply src/index.ts:1842]: Although the complexity here might increase if we're resolving other namesystems too (e.g. unstoppable domains, beam names, coinbase ids, etc)

File: test/basic/resolveToENSName.test.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
+import * as peanut from '../../src/index'
+
+describe('resolveToENSName', () => {
+	it('should resolve ENS name when using default provider', async () => {
+		const address = '0x7fDCc4908bEF4f6F296f78a30A6b295069EE9e5d'
+		const ensName = 'borcherd.eth'
+		const result = await peanut.resolveToENSName({ address })
+		console.log(result)
+		expect(result).toBe(ensName)
+	}, 60000)
```
Comment (Hugo0) [inline test/basic/resolveToENSName.test.ts:10]: Now you have to keep renewing that ens name or this test will fail ahaha

File: test/basic/getLinkDetails.test.ts [hunk 4]
```diff
@@ -84,6 +84,17 @@ describe('getLinkDetails', function () {
 			console.log(linkDetails.depositDate)
 		}, 1000000)
 
+		it.only('should include senderAddress', async function () {
```
Comment (Hugo0) [inline test/basic/getLinkDetails.test.ts:87]: Do not commit .only in test files!

File: src/index.ts (hunk 2 above)
Comment (borcherd) [reply src/index.ts:1842]: Yep, this was my thinking too. The integrator does not have to worry about what nameSystem & provider to use, just passin the address and get the name back

File: test/basic/getLinkDetails.test.ts (hunk 4 above)
Comment (borcherd) [reply test/basic/getLinkDetails.test.ts]: Yep, sorry!

**Pull request merged**